### PR TITLE
Flakiness combinators

### PR DESCRIPTION
**Pull request by NorfairKing**
CC @vidocco

### CODE REVIEW

**Review by CSVdB — APPROVED** [review]
Lgtm

File: sydtest/src/Test/Syd/Modify.hs [hunk 1]
```diff
@@ -65,3 +71,19 @@ doNotRandomiseExecutionOrder = withExecutionOrderRandomisation DoNotRandomiseExe
...
 withExecutionOrderRandomisation :: ExecutionOrderRandomisation -> TestDefM a b c -> TestDefM a b c
 withExecutionOrderRandomisation p = censor ((: []) . DefRandomisationNode p)
+
+-- | Mark a test suite as "potentially flaky".
+--
+-- This will retry any test in the given test group up to the given number of tries, and pass a test if it passes once.
+-- The test output will show which tests were flaky.
+--
+-- WARNING: This is only a valid approach to dealing with test flakiness if it is true that tests never pass accidentally.
+-- In other words: it must be true that a true positive test failure will fail every time.
```
Comment (vidocco) [inline sydtest/src/Test/Syd/Modify.hs:81]: Nit, I'd probably reword this to something like (current wording makes it a bit confusing to read):
`In other words: tests using flaky must be guaranteed to fail every time if an error is introduced in the code, it should only be added to deal with accidental failures, never accidental passes.`

**Review by vidocco — APPROVED** [review]
tiny nit about wording, other than that LGTM!
